examples/neural_compressor/question-answering/run_qa.py

```diff
@@ -513,6 +545,87 @@ def prepare_train_features(examples):
             # Number of samples might increase during Feature Creation, We select only specified max samples
             train_dataset = train_dataset.select(range(data_args.max_train_samples))
 
+    accelerator = Accelerator(cpu=training_args.no_cuda)
+
+    def move_input_to_device(input, device):
+        if isinstance(input, torch.Tensor):
+            return input if input.device == device else input.to(device)
+        elif isinstance(input, tuple):
+            return tuple([move_input_to_device(ele, device) for ele in input])
+        elif isinstance(input, list):
+            return [move_input_to_device(ele, device) for ele in input]
+        elif isinstance(input, dict):
+            return {key: move_input_to_device(input[key], device) for key in input}
+        else:
+            raise TypeError("Only inputs types torch.Tensor, tuple, list and dict are supported")
+
+    # get logits of teacher model
+    # declare teacher config and model for distillation
+    teacher_config = None
+    teacher_model = None
+    if optim_args.generate_teacher_logits:
+        if not data_args.pad_to_max_length:
+            raise ValueError("To computes teacher logits, pad_to_max_length must be set to True")
+        teacher_config = AutoConfig.from_pretrained(optim_args.teacher_model_name_or_path)
+        teacher_model = AutoModelForQuestionAnswering.from_pretrained(
+            optim_args.teacher_model_name_or_path,
+            from_tf=bool(".ckpt" in optim_args.teacher_model_name_or_path),
+            config=teacher_config,
+        )
+        teacher_model_qa = QAModel(teacher_model)
+        teacher_model_qa = accelerator.prepare(teacher_model_qa)
+        num_param = lambda model: sum(p.numel() for p in model.parameters())
+        logger.info(
+            "***** Number of teacher model parameters: {:.2f}M *****".format(num_param(teacher_model_qa) / 10**6)
+        )
+        logger.info("***** Number of student model parameters: {:.2f}M *****".format(num_param(model) / 10**6))
+
+        def get_logits(teacher_model_qa, train_dataset):
+            logger.info("***** Getting logits of teacher model *****")
+            logger.info(f"  Num examples = {len(train_dataset) }")
+            logger.info(f"  Batch Size = {training_args.per_device_eval_batch_size }")
+            npy_file = os.path.join(
+                os.path.dirname(os.path.abspath(__file__)),
+                "{}.{}.npy".format(data_args.dataset_name, optim_args.teacher_model_name_or_path.replace("/", ".")),
+            )
+            if os.path.exists(npy_file):
```

Comment:
I would be in favor of removing the saving / loading of the teacher logits as the training dataset could vary in length or be preprocessed differently from one run to another